app/src/main/java/com/example/makeitso/model/service/impl/AccountServiceImpl.kt

```diff
@@ -17,57 +17,66 @@ limitations under the License.
 package com.example.makeitso.model.service.impl
 
 import com.example.makeitso.model.service.AccountService
+import com.example.makeitso.model.service.User
+import com.example.makeitso.model.service.trace
 import com.google.firebase.auth.EmailAuthProvider
+import com.google.firebase.auth.FirebaseAuth
 import com.google.firebase.auth.ktx.auth
 import com.google.firebase.ktx.Firebase
+import kotlinx.coroutines.channels.awaitClose
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.callbackFlow
+import kotlinx.coroutines.tasks.await
 import javax.inject.Inject
 
-class AccountServiceImpl @Inject constructor() : AccountService {
-    override fun hasUser(): Boolean {
-        return Firebase.auth.currentUser != null
-    }
-
-    override fun isAnonymousUser(): Boolean {
-        return Firebase.auth.currentUser?.isAnonymous ?: true
-    }
+private const val CREATE_ACCOUNT_TRACE = "createAccount"
```

Comment:
Can we put this in a companion object inside the class? (Like we do in `ConfigurationServiceImpl`).